src/main/scala/com/activegrid/model/Software.scala

```diff
@@ -0,0 +1,65 @@
+package com.activegrid.model
+
+import com.activegrid.utils.ActiveGridUtils
+import com.typesafe.scalalogging.Logger
+import org.neo4j.graphdb.{Node, NotFoundException}
+import org.slf4j.LoggerFactory
+
+/**
+  * Created by sampathr on 22/9/16.
+  */
+case class Software(override val id: Option[Long],
+                    version: Option[String],
+                    name: String,
+                    provider: String,
+                    downloadURL: Option[String],
+                    port: String,
+                    processNames: List[String],
+                    discoverApplications: Boolean) extends BaseEntity
+
+object Software {
+  implicit class SoftwareImpl(software: Software) extends Neo4jRep[Software] {
+    val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+    val label = "SoftwaresTest2"
+
+    override def toNeo4jGraph(software: Software): Option[Node] = {
+      logger.debug(s"In toGraph for Software: $software")
+      val map = Map("version" -> software.version.getOrElse(None),
+        "name" -> software.name,
+        "provider" -> software.provider,
+        "downloadURL" -> software.downloadURL.getOrElse(None),
+        "port" -> software.port,
+        "processNames" -> software.processNames.toArray,
+        "discoverApplications" -> software.discoverApplications)
+
+      val softwareNode = GraphDBExecutor.saveEntity[Software](label, map)
+      Some(softwareNode)
+    }
+
+    override def fromNeo4jGraph(nodeId: Long): Option[Software] = {
+      Software.fromNeo4jGraph(nodeId)
+    }
+
+  }
+
+  def fromNeo4jGraph(nodeId: Long): Option[Software] = {
+    try {
+      val node = GraphDBExecutor.findNodeById(nodeId)
+      val map = GraphDBExecutor.getProperties(node.get, "version", "name", "provider", "downloadURL", "port", "processNames", "discoverApplications")
+
+      val software = Software(Some(nodeId),
+        ActiveGridUtils.getValueFromMapAs[String](map, "version"),
+        map("name").asInstanceOf[String],
+        map("provider").asInstanceOf[String],
+        ActiveGridUtils.getValueFromMapAs[String](map, "downloadURL"),
+        map("port").asInstanceOf[String],
+        map("processNames").asInstanceOf[Array[String]].toList,
+        map("discoverApplications").asInstanceOf[Boolean])
+      Some(software)
+    } catch {
+      case nfe: NotFoundException => None
+      case exception: Exception => throw new Exception("Unable to get the Entity")
```

Comment:
Just don't catch the exception and the caller will log the original exception with proper stack trace.